datasette/app.py

```diff
@@ -164,9 +164,16 @@ def __init__(
         memory=False,
         config=None,
         version_note=None,
+        config_dir=None,
     ):
+        assert config_dir is None or isinstance(
+            config_dir, Path
+        ), "config_dir= should be a pathlib.Path"
+        # TODO: Use 'inspect-data.json' to decide on immutables
         immutables = immutables or []
```

Comment:
If `inspect-data.json` is present and a database file is listed there, treat that file as immutable.